File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
class Solution {
    public long solution(long n) {
        long answer = -1;
        long varNum= (long) Math.sqrt(n);
        if (Math.sqrt(n)%1 == 0){
            varNum=varNum+1;
            answer=varNum*varNum;
        }
        return answer;
    }
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.kt
import kotlin.math.*

class Solution {
    fun solution(n: Long): Long {
        var answer: Long = 0
        var m=n.toDouble()
        var x:Double=sqrt(m)
        if(x%1.0 !=0.0){
            x=-1.0
            answer=x.toLong()
        }
        else if(x%1.0 == 0.0){
            x=x+1
            answer=(x*x).toLong()
        }
        return answer
    }
}